scripts/oecd/regional_demography/deaths/preprocess_csv.py

```diff
@@ -13,120 +13,115 @@
 # limitations under the License.
 
 import sys
-
-sys.path.append('../')
-from utils import multi_index_to_single_index
+import os
 import csv
 import json
 import pandas as pd
+from absl import flags, logging
+from absl import app
+from columns import *
+
+_FLAGS = flags.FLAGS
+_MODULE_DIR = os.path.dirname(os.path.abspath(__file__))
+parent_dir = os.path.dirname(_MODULE_DIR)
+flags.DEFINE_string('mode', '', 'Options: download or process')
+
+_MODULE_DIR = os.path.dirname(os.path.abspath(__file__))
+
+sys.path.insert(1, os.path.join(_MODULE_DIR, '../'))
+from utils import multi_index_to_single_index
+from download import download_data_to_file_and_df
+
+
+def process_data(df, output_file_path):
+    logging.info("Processing the input file")
+    logging.info("Filtering the required columns")
+    df1 = df[(df['TERRITORIAL_LEVEL'] == "TL3")]
+    df1 = df[[
+        'TERRITORIAL_LEVEL', 'REF_AREA', 'Reference area', 'AGE', 'SEX',
+        'TIME_PERIOD', 'OBS_VALUE', 'UNIT_MEASURE'
+    ]]
+    df1.loc[:, "AGE"] = df1["UNIT_MEASURE"] + "_" + df1["AGE"] + df1["SEX"]
+
+    # First remove geos with names that we don't have mappings to dcid for.
+    try:
+        regid_file = os.path.join(parent_dir, "regid2dcid.json")
+        with open(regid_file, 'r') as f:
+            regid2dcid = dict(json.loads(f.read()))
+        logging.info("Resolving places")
+        df1 = df1[df1['REF_AREA'].isin(regid2dcid.keys())]
```

Comment:
Can we log a count of how many were resolved, unresolved?

Reply:
Added a log for number of resolved and unresolved places.